synapse/rest/media/v1/thumbnailer.py

```diff
@@ -30,6 +30,26 @@ class Thumbnailer(object):
 
     def __init__(self, input_path):
         self.image = Image.open(input_path)
+        try:
+            # We don't use ImageOps.exif_transpose since it crashes with big EXIF
+            image_exif = self.image._getexif()
+            image_orientation = image_exif.get(0x0112)
+            transpose_method = {
+                2: Image.FLIP_LEFT_RIGHT,
+                3: Image.ROTATE_180,
+                4: Image.FLIP_TOP_BOTTOM,
+                5: Image.TRANSPOSE,
+                6: Image.ROTATE_270,
+                7: Image.TRANSVERSE,
+                8: Image.ROTATE_90
+            }.get(image_orientation)
+            if transpose_method is not None:
+                self.image = self.image.transpose(transpose_method)
+                # We don't need EXIF any more
+                self.image.info["exif"] = None
```

Comment:
this seems a bit brutal?

Reply:
In class Thumbnailer the main goal is to generate a Thumbnail.
Thumbnails normally don't have any EXIF information since the main goal of them are to weight less.

The current released version of this class is generating thumbnails without rotating and is droping all EXIF information (including the orientation tag) and this is what is causing the issue.

So since all EXIF information is currently droped when exporting the scaled image, we don't really need this information at all either for current code nor for this pull request code.

This line is only to avoid any future issue with the wrong orientation EXIF tag, because in this case the image is already rotated and the EXIF orientation tag would tell that needs to rotate.

The nice solution would be trying to remove only EXIF orientation tag.
This is what ImageOps.exif_transpose is trying to do but crashes with images with big EXIF (generated by Apple iPhone for example).

To summarize:
- Curent Thumbnailer class is dropping EXIF on export image.
- EXIF information is not needed once the image has been rotated.
- EXIF information can be droped for all images as soon as is loaded and rotated to free memory sooner, but this would make code more strange.

Reply:
You're right.

When I first looked at this code, I assumed that we were stripping the EXIF information from the *original* image. (There may be valid reasons to do that, but it's a thing that we'd need to think about.) However, I now see that the original image is not affected. So this is fine :+1: